features/netsocket/TCPSocket.cpp

```diff
@@ -79,6 +79,10 @@ nsapi_error_t TCPSocket::connect(const SocketAddress &address)
 
     _write_in_progress = false;
 
+    if (ret == NSAPI_ERROR_OK) {
```

Comment:
This needs to be below the following IS_CONNECTED->OK conversion.

Reply:
But is it then creating a loophole where you first set socket to non blocking mode. Then call `connect(A)` and later `connect(B)` would actually change the connection address here, even that the actual connection is with peer `A`?

I'm unsure whether IP stacks actually check the connection addresses after a first call.

Reply:
Okay, but in non-blocking there's possibly never an "OK" - it's "INPROGESS", "ALREADY", "ISCONN". If you don't want to note it on "ISCONN" then note it on "INPROGRESS". That should be the return code for "I've accepted these parameters and am acting on them, but I've not finished yet".